### PR TITLE
Created Edit User route in Flask API

### LINKED ISSUE #534: Create Edit User route in Flask API
**Is your feature request related to a problem? Please describe.**

The Flask API lacks the Edit User route which the NodeJS API has. Due to this, the edit user functionality is not working in the React Frontend when run with the flask server.

### PR DESCRIPTION
**Pull request by amlannandy**
# Description

The edit user route which was present in the NodeJS API was missing in the Flask API. I created the route for the same, which sends the exact same response so that the Flask API works with the React Frontend without any changes in the frontend code.

Fixes #534 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested using Web App and Postman Client

screen-capture

**Test Configuration**:

Google Chrome, Windows 10

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by amlannandy** [comment]
@carpecodeum Hello! Could you please review this?

File: labellab-flask/api/controllers/userscontroller.py [hunk 1]
```diff
@@ -382,6 +383,74 @@ def get(self,query):
...
+            response = {
+                "success": False,
+                "msg": 'Please provide email/username to update'
+            }
+            return make_response(jsonify(response)), 400
+
+        # if some data is sent but it doesn't contain email or username
+        try:
+            username = put_data["username"]
+            email = put_data["email"]
```
Comment (carpecodeum) [inline labellab-flask/api/controllers/userscontroller.py:416]: use the .get() method here to check empty quotes

File: labellab-flask/api/controllers/userscontroller.py [hunk 2]
```diff
@@ -382,6 +383,74 @@ def get(self,query):
...
+            }
+            return make_response(jsonify(response)), 400
+
+        existing_user = find_by_email(email)
+
+        if existing_user and email != user['email']:
+            # There is an existing user with that email. So we can't let the current user use this email
+            response = {
+                "success": False,
+                "msg": "New email already associated with another account"
```
Comment (carpecodeum) [inline labellab-flask/api/controllers/userscontroller.py:430]: use a simpler response message

**Comment by carpecodeum** [comment]
@amlannandy Thanks for this PR I have some reservations about being able to change the email of an existing user will that not make the existing older email redundant data in the database? Due to this, I don't see the point of editing the email of a user.

**Comment by carpecodeum** [comment]
Also do submit a Gif of the function working as well

**Comment by amlannandy** [comment]
> @amlannandy Thanks for this PR I have some reservations about being able to change the email of an existing user will that not make the existing older email redundant data in the database? Due to this, I don't see the point of editing the email of a user.

@carpecodeum Actually, after the change the old email no longer exists in the database and can be used for opening a new account. So I don't think there is redundancy. Should I still remove the feature? 

Also, I'm working on the other changes you asked for and will update that with the GIF soon. 👍🏻

**Comment by carpecodeum** [comment]
> > @amlannandy Thanks for this PR I have some reservations about being able to change the email of an existing user will that not make the existing older email redundant data in the database? Due to this, I don't see the point of editing the email of a user.
> 
> @carpecodeum Actually, after the change the old email no longer exists in the database and can be used for opening a new account. So I don't think there is redundancy. Should I still remove the feature?
> 
> Also, I'm working on the other changes you asked for and will update that with the GIF soon. 👍🏻

No I dont think this would be very practical as any user who wants to change their email would already delete their account and then make a new one. there is a reason why most of the good web apps dont have a feature to change the email of the user which was insteadf used to log in the user in the first place.

**Comment by amlannandy** [comment]
> > > @amlannandy Thanks for this PR I have some reservations about being able to change the email of an existing user will that not make the existing older email redundant data in the database? Due to this, I don't see the point of editing the email of a user.
> > 
> > 
> > @carpecodeum Actually, after the change the old email no longer exists in the database and can be used for opening a new account. So I don't think there is redundancy. Should I still remove the feature?
> > Also, I'm working on the other changes you asked for and will update that with the GIF soon. 👍🏻
> 
> No I dont think this would be very practical as any user who wants to change their email would already delete their account and then make a new one. there is a reason why most of the good web apps dont have a feature to change the email of the user which was instead used to log in the user in the first place.

Okay, I'll remove it then. 👍🏻

**Comment by amlannandy** [comment]
@carpecodeum Made the changes 👍🏻

**Review by carpecodeum — APPROVED** [review]

**Pull request merged**